eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1743,4 +1750,53 @@ private long computeStalenessMonitorDelay(long delay) {
         }
     }
 
+    /**
+     * Gets stats for the DiscoveryClient.
+     *
+     * @return The DiscoveryClientStats instance.
+     */
+    public Stats getStats() {
+        return stats;
+    }
+
+    /**
+     * Stats is used to track useful attributes of the DiscoveryClient. It includes helpers that can aid
+     * debugging and log analysis.
+     */
+    public class Stats {
+
+        private Stats() {}
+
+        public int initLocalRegistrySize() {
+            return initRegistrySize;
+        }
+
+        public long initTimestamp() {
```

Comment:
nit: initTimestampMs() to disambiguate the unit.

Reply:
This would make it consistent with the private field. Do you think we should also use `lastSuccessfulRegistryFetchTimestampMs` and `lastSuccessfulHeartbeatTimestampMs` to be consistent? We could do that with stats and leave the private fields as-is.

Reply:
Yes good point, agree with you on both accounts, rename the stat ones to include the unit, leave the fields (or you can rename those as well if you prefer so).

Reply:
Done. Left the private fields as-is.